src/Microsoft.Security.Utilities.Core/IdentifiableScan.cs

```diff
@@ -2,141 +2,64 @@
-
-    [DllImport("microsoft_security_utilities_core")]
-    static extern UInt32 identifiable_scan_def_count(IntPtr scan);
-
-    [DllImport("microsoft_security_utilities_core")]
-    static extern void identifiable_scan_def_name(
-        IntPtr scan,
-        UInt32 index,
-        byte[] name,
-        ref long nameLength);
-
-    [DllImport("microsoft_security_utilities_core")]
-    static extern bool identifiable_scan_match_get(
-        IntPtr scan,
-        UInt32 index,
-        out UInt64 start,
-        out UInt64 len);
-
-    [DllImport("microsoft_security_utilities_core")]
-    static extern bool identifiable_scan_match_check(
-        IntPtr scan,
-        UInt32 index,
-        byte[] input,
-        long inputLength,
-        out UInt32 defIndex,
-        byte[] output,
-        ref long outputLength);
-
     private readonly Dictionary<string, IList<RegexPattern>> signatureToPatternsMap;
-    private readonly Dictionary<string, ISet<string>> idToSignaturesMap;
+    private readonly HighPerformanceScanner highPerformanceSecretScanner;
     private SecretMasker backupSecretMasker;
-    private IRegexEngine regexEngine;
-    private List<string> orderedIds;
-    private IntPtr scan;
-
-    private static readonly ISet<string> HighPerformanceEnabledSignatures = new HashSet<string>(new string[]
-        {
-            "+ASt",
-            "AzCa",
-            "7Q~",
-            "8Q~",
-            "AzFu",
-            "ACDb",
-            "+ABa",
-            "AzSe",
-            "AzSe",
-            "+AMC",
-            "+ASb",
-            "+AEh",
-            "+ARm",
-            "+ACR",
-            "+AIoT",
-            "APIM",
-            "AZEG",
-            "JQQJ",
-        });
+    private readonly IRegexEngine regexEngine;
+    private readonly List<string> orderedIds;
 
     public IdentifiableScan(IEnumerable<RegexPattern> regexPatterns, bool generateCorrelatingIds, IRegexEngine regexEngine = null)
-    {   
+    {
         this.signatureToPatternsMap = new Dictionary<string, IList<RegexPattern>>();
-        this.idToSignaturesMap = new Dictionary<string, ISet<string>>();
         this.regexEngine = regexEngine ?? CachedDotNetRegex.Instance;
         this.generateCorrelatingIds = generateCorrelatingIds;
         this.orderedIds = new List<string>();
 
+        var highPerformancePatterns = new HashSet<CompiledHighPerformancePattern>();
+        
         foreach (RegexPattern pattern in regexPatterns)
         {
             if (pattern.Signatures == null)
             {
                 PopulateBackupMasker(generateCorrelatingIds);
-
                 this.backupSecretMasker.AddRegex(pattern);
                 continue;
             }
 
             foreach (string signature in pattern.Signatures)
             {
-                if (!HighPerformanceEnabledSignatures.Contains(signature))
+                var highPerformancePattern = CompiledHighPerformancePattern.ForSignature(signature);
```

Comment:
>CompiledHighPerformancePattern [](http://example.com/codeflow?start=45&length=30)

i'm surprised `var` is permitted here as the expressed type of the local isn't apparent from the remaining code pattern.

Reply:
It's because I deleted stylecop which broke on new patterns. I'll put the enforcement back with dotnet format. I'll make a different PR for that. See also other comment on var, which was meant to be my last, but then I replied on this related detail.

Reply:
Actually maybe not why because I see some other pre-exiting vars that don't have obvious type. But regardless, once I match cask formatting check here, this will be enforced.